From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: kierandrewett <kieran@drewett.dev>
Subject: [PATCH] preserve popup client origin

---
--- a/src/core/constraints.c
+++ b/src/core/constraints.c
@@ -39,6 +39,7 @@
 #include "core/workspace-private.h"
 #include "meta/meta-external-constraint.h"
 #include "meta/prefs.h"
+#include "wayland/meta-gnoblin-window-frame.h"
 
 #ifdef HAVE_XWAYLAND
 #include "x11/meta-x11-frame.h"
@@ -951,6 +952,7 @@
 
   parent = meta_window_get_transient_for (window);
   parent_rect = meta_window_config_get_rect (parent->config);
+  meta_gnoblin_window_frame_rect_to_client (parent, NULL, &parent_rect);
   if (window->placement.state == META_PLACEMENT_STATE_CONSTRAINED_FINISHED)
     {
       placement_rule->parent_rect.x = parent_rect.x;
--- a/src/wayland/meta-window-wayland.c
+++ b/src/wayland/meta-window-wayland.c
@@ -1730,6 +1730,7 @@
 
               parent = meta_window_get_transient_for (window);
               parent_rect = meta_window_config_get_rect (parent->config);
+              meta_gnoblin_window_frame_rect_to_client (parent, NULL, &parent_rect);
               rect.x = parent_rect.x + acked_configuration->rel_x;
               rect.y = parent_rect.y + acked_configuration->rel_y;
             }
--- a/src/wayland/meta-wayland-xdg-shell.c
+++ b/src/wayland/meta-wayland-xdg-shell.c
@@ -2555,9 +2555,9 @@
               if (configuration->has_size)
                 {
                   parent_rect.width =
-                    configuration->width / configuration->scale;
+                    configuration->width;
                   parent_rect.height =
-                    configuration->height / configuration->scale;
+                    configuration->height;
                 }
               meta_gnoblin_window_frame_rect_to_client (parent_window,
                                                          configuration,
@@ -2570,8 +2570,8 @@
               meta_rectangle_resize_with_gravity (&parent_rect,
                                                   &parent_rect,
                                                   configuration->gravity,
-                                                  xdg_positioner->parent_width,
-                                                  xdg_positioner->parent_height);
+                                                  xdg_positioner->parent_width * meta_window_wayland_get_geometry_scale (parent_window),
+                                                  xdg_positioner->parent_height * meta_window_wayland_get_geometry_scale (parent_window));
             }
           else
             {
@@ -2592,8 +2592,8 @@
       meta_rectangle_resize_with_gravity (&parent_rect,
                                           &parent_rect,
                                           META_GRAVITY_SOUTH_EAST,
-                                          xdg_positioner->parent_width,
-                                          xdg_positioner->parent_height);
+                                          xdg_positioner->parent_width * meta_window_wayland_get_geometry_scale (parent_window),
+                                          xdg_positioner->parent_height * meta_window_wayland_get_geometry_scale (parent_window));
     }
   else
     {
